core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -249,7 +249,14 @@ private[spark] class ExecutorAllocationManager(
    * yarn-client mode when AM re-registers after a failure.
    */
   def reset(): Unit = synchronized {
-    initializing = true
+    /**
+     * When some tasks need to be scheduled, resetting the initializing field may cause
+     * it to not be set to false in yarn.
```

Comment:
Currently this method will only be called in yarn-client mode when AM re-registers after a failure.